### PR TITLE
feat: Introducing WholesaleResultQueries class

### PR DESCRIPTION
**Pull request by JesperGomes**

### CODE REVIEW

**Review by dstenroejl — COMMENTED** [review]
Added a few comments

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    private readonly DatabricksSqlStatementApiFixture _fixture;
+
+    public WholesaleResultQueriesTests(DatabricksSqlStatementApiFixture fixture)
+    {
+        _fixture = fixture;
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task GetAsync_ReturnsExpectedWholesaleResult(
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs:51]: Missing "When" statement in name?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/CalculationResults/IWholesaleResultQueries.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model.WholesaleResults;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults;
+
+public interface IWholesaleResultQueries
+{
+    /// <summary>
+    /// Get all wholesale results for a given batch.
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/CalculationResults/IWholesaleResultQueries.cs:22]: changed

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            calculationResultId: MonthlyAmountTariffCalculationResultId,
+            chargeType: "tariff",
+            chargeResolution: "P1M",
+            meteringPointType: null,
+            settlementMethod: null,
+            amount: DefaultMonthlyAmount);
+
+        var rows = new List<IReadOnlyCollection<string?>> { hourlyTariffRow, monthlyAmountTariffRow };
+        await _fixture.DatabricksSchemaManager.InsertAsync<WholesaleResultColumnNames>(options.Value.WHOLESALE_RESULTS_TABLE_NAME, rows);
+    }
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs:106]: We could drop `options` parameter and just acccess the `_fixture` (if we do the code callling this method must be updated too; notice I have postfixed method name with Async):

```suggestion
    private async Task InsertHourlyTariffAndMonthlyAmountTariffRowsAsync()
    {
        var hourlyTariffRow = WholesaleResultDeltaTableHelper.CreateRowValues(
            calculationId: CalculationId,
            calculationResultId: HourlyTariffCalculationResultId,
            chargeType: "tariff",
            chargeResolution: "PT1H",
            meteringPointType: "production",
            settlementMethod: null,
            amount: DefaultHourlyAmount);

        var monthlyAmountTariffRow = WholesaleResultDeltaTableHelper.CreateRowValues(
            calculationId: CalculationId,
            calculationResultId: MonthlyAmountTariffCalculationResultId,
            chargeType: "tariff",
            chargeResolution: "P1M",
            meteringPointType: null,
            settlementMethod: null,
            amount: DefaultMonthlyAmount);

        var rows = new List<IReadOnlyCollection<string?>> { hourlyTariffRow, monthlyAmountTariffRow };

        var wholesaleResultTableName = _fixture.DatabricksSchemaManager.DeltaTableOptions.Value.WHOLESALE_RESULTS_TABLE_NAME
        await _fixture.DatabricksSchemaManager.InsertAsync<WholesaleResultColumnNames>(wholesaleResultTableName, rows);
    }
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        var actualHourlyAmount = actual.Single(row => row.Id.ToString() == HourlyTariffCalculationResultId);
+        actualHourlyAmount.ChargeResolution.Should().Be(ChargeResolution.Hour);
+        actualHourlyAmount.TimeSeriesPoints.First().Amount.Should().Be(decimal.Parse(DefaultHourlyAmount, CultureInfo.InvariantCulture));
+
+        var actualMonthlyAmount = actual.Single(row => row.Id.ToString() == MonthlyAmountTariffCalculationResultId);
+        actualMonthlyAmount.ChargeResolution.Should().Be(ChargeResolution.Month);
+        actualMonthlyAmount.TimeSeriesPoints.First().Amount.Should().Be(decimal.Parse(DefaultMonthlyAmount, CultureInfo.InvariantCulture));
+    }
+
+    private async Task InsertHourlyTariffAndMonthlyAmountTariffRows(IOptions<DeltaTableOptions> options)
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs:84]: Should this method name be postfixed with *Async?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        // Arrange
+        var deltaTableOptions = _fixture.DatabricksSchemaManager.DeltaTableOptions;
+        await InsertHourlyTariffAndMonthlyAmountTariffRows(deltaTableOptions);
+        batch = batch with { BatchId = Guid.Parse(CalculationId) };
+        var sqlStatementClient = _fixture.CreateSqlStatementClient(
+            httpClientFactoryMock,
+            sqlStatusResponseParserLoggerMock,
+            databricksSqlStatementClientLoggerMock);
+        batchesClientMock.Setup(b => b.GetAsync(It.IsAny<Guid>())).ReturnsAsync(batch);
+        var sut = new WholesaleResultQueries(sqlStatementClient, batchesClientMock.Object, deltaTableOptions, wholesaleResultQueriesLoggerMock.Object);
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs:68]: We could split this line and access options directly (if we also made the other suggested refactoring of `InsertHourlyTariffAndMonthlyAmountTariffRows` we can remove the variable `deltaTableOptions`):
```suggestion
        var sut = new WholesaleResultQueries(
            sqlStatementClient, 
            batchesClientMock.Object, 
            _fixture.DatabricksSchemaManager.DeltaTableOptions, 
            wholesaleResultQueriesLoggerMock.Object);
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        // Arrange
+        var deltaTableOptions = _fixture.DatabricksSchemaManager.DeltaTableOptions;
+        await InsertHourlyTariffAndMonthlyAmountTariffRows(deltaTableOptions);
+        batch = batch with { BatchId = Guid.Parse(CalculationId) };
+        var sqlStatementClient = _fixture.CreateSqlStatementClient(
+            httpClientFactoryMock,
+            sqlStatusResponseParserLoggerMock,
+            databricksSqlStatementClientLoggerMock);
+        batchesClientMock.Setup(b => b.GetAsync(It.IsAny<Guid>())).ReturnsAsync(batch);
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs:67]: Would it be easier to read our mock setup if we always split the lines?

```suggestion
        batchesClientMock
            .Setup(b => b.GetAsync(It.IsAny<Guid>()))
            .ReturnsAsync(batch);
```

**Review by dstenroejl — COMMENTED** [review]
Final comments

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SqlStatements/DeltaTableConstants/WholesaleResultColumnNames.cs [hunk 7]
```diff
@@ -12,12 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System.Reflection;
+
 namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements.DeltaTableConstants;
 
 public class WholesaleResultColumnNames
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SqlStatements/DeltaTableConstants/WholesaleResultColumnNames.cs:19]: Then I will get issues in a test that does this:
InsertAsync<EnergyResultColumnNames>(...)

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs (hunk 6 above)
Comment (JesperGomes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/WholesaleResultQueriesTests.cs:67]: Maybe. I can do it :-)

**Review by dstenroejl — APPROVED** [review]
LGTM

**Pull request merged**